st2reactor/st2reactor/rules/datatransform.py

```diff
@@ -60,7 +60,7 @@ def _construct_system_context(data, context):
         """Identify the system context in the data."""
         # The following regex will look for all occurrences of "{{system.*}}",
         # "{{ system.* }}", "{{ system.*}}", and "{{system.* }}" in the data.
-        regex = '{{\s*' + SYSTEM_PREFIX + '.(.*?)\s*}}'
+        regex = '{{\s*' + SYSTEM_KV_PREFIX + '.(.*?)\s*}}'
```

Comment:
Fixed in c360723a3565d7dacd96434a53017471db2a94d1. No more regex; replaced with KeyValueLookup. This way we can support dotted properties.